src/main/scala/sigmastate/eval/Evaluation.scala

```diff
@@ -349,9 +358,12 @@ trait Evaluation extends RuntimeCosting { IR =>
           te.sym.getMetadata(OperationIdKey) match {
             case Some(opId: OperationId) =>
               if (opId.opType.tRange.isCollection) {
-                val col = res._1(res._2).asInstanceOf[special.collection.Col[Any]]
-                val colTime = if (col.length > 1) estimatedTime / col.length else estimatedTime
-                CostTableStat.addOpTime(opId, colTime, col.length)
+                res._1(res._2) match {
```

Comment:
@ergomorphic Good catch! Thanks. It was due to incorrect result packing for `SubstConstansts` as a constant instead of a collection. Fixed in https://github.com/ScorexFoundation/sigmastate-interpreter/pull/334 and made the expression in evaluation test closer to ergo (added `EQ`).